internal/fsutil/path.go

```diff
@@ -0,0 +1,47 @@
+package fsutil
+
+import (
+	"path/filepath"
+	"strings"
+)
+
+// These functions exists because we work with slash terminated directory paths
+// that come from deb package tarballs but standard library path functions
+// treat slash terminated paths as unclean.
+
+// Like filepath.Dir() but trailing slash on input is ignored and the return
+// value always includes trailing slash.
+//
+// Comparison of filepath.Dir() with fsutil.Dir():
+//
+//	filepath.Dir("/foo/bar/")	== "/foo/bar"
+//	filepath.Dir("/foo/bar")	== "/foo"
+//
+//	fsutil.Dir("/foo/bar")		== "/foo/"
+//	fsutil.Dir("/foo/bar/")		== "/foo/"
+func Dir(path string) string {
+	parent := filepath.Dir(filepath.Clean(path))
+	if parent != "/" {
+		parent += "/"
+	}
+	return parent
+}
+
+// Like filepath.Clean() but keeps trailing slash.
```

Comment:
Rather than directing people to documentation for `filepath.Clean()` it could be better to write exactly what it does here. (they may not necceserily know exactly what `filepath.Clean` does.)

Reply:
Replied above. Resolving this one.